File: 605-flowers/solution.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        r = 0
        q = 0
        for i in flowerbed:
            if q == 3 or (r > 0 and q == 3):
                r += 1
                q = 0
            q += 1
            if flowerbed[i] == 1:
                q = 0
        return r >= n

File: 605-flowers/solution.go
func canPlaceFlowers(flowerbed []int, n int) bool {
	size := len(flowerbed)
	r := 0
	for idx, i := range flowerbed {
		var hasNextZ bool = func() bool {
			if idx == size-1 || flowerbed[idx+1] == 0 {
				return true
			}
			return false
		}()
		var hasPrevZ bool = func() bool {
			if idx == 0 || flowerbed[idx-1] == 0 {
				return true
			}
			return false
		}()

		if i == 0 && hasNextZ && hasPrevZ {
			r += 1
			flowerbed[idx] = 1
		}
	}

	return r >= n
}